staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -200,22 +201,28 @@ func getNestedField(obj map[string]interface{}, fields ...string) interface{} {
 	return val
 }
 
-func getNestedString(obj map[string]interface{}, fields ...string) string {
-	if str, ok := getNestedField(obj, fields...).(string); ok {
+// NestedString returns the string value of a nested field.
+// Returns empty string if value is not found or is not a string.
+func NestedString(obj map[string]interface{}, fields ...string) string {
+	if str, ok := NestedField(obj, fields...).(string); ok {
 		return str
 	}
 	return ""
 }
 
-func getNestedInt64(obj map[string]interface{}, fields ...string) int64 {
-	if str, ok := getNestedField(obj, fields...).(int64); ok {
+// NestedInt64 returns the int64 value of a nested field.
+// Returns 0 if value is not found or is not an int64.
+func NestedInt64(obj map[string]interface{}, fields ...string) int64 {
+	if str, ok := NestedField(obj, fields...).(int64); ok {
 		return str
 	}
 	return 0
 }
 
-func getNestedInt64Pointer(obj map[string]interface{}, fields ...string) *int64 {
-	nested := getNestedField(obj, fields...)
+// NestedInt64Pointer returns the *int64 value of a nested field.
+// Returns nil if value is not found or is not an int64/*int64.
+func NestedInt64Pointer(obj map[string]interface{}, fields ...string) *int64 {
```

Comment:
why a pointer and not the value?

Reply:
The other funcs return values or clones, never the original object or a reference to it.